src/runtime/logging.cc

```diff
@@ -166,10 +167,118 @@ namespace tvm {
 namespace runtime {
 namespace detail {
 
+namespace {
+constexpr const char* kSrcPrefix = "/src/";
+constexpr const size_t kSrcPrefixLength = 5;
+constexpr const char* kDefaultKeyword = "DEFAULT";
+}  // namespace
+
+// Parse \p opt_spec as a VLOG specification as per comment in
+// DebugLoggingEnabled and VerboseLoggingEnabled.
+std::unordered_map<std::string, int> ParseTvmLogDebugSpec(const char* opt_spec) {
+  std::unordered_map<std::string, int> map;
+  if (opt_spec == nullptr) {
+    // DLOG and VLOG disabled.
+    return map;
+  }
+  std::string spec(opt_spec);
+  if (spec.empty() || spec == "0") {
+    // DLOG and VLOG disabled.
+    return map;
+  }
+  if (spec == "1") {
+    // Legacy specification for enabling just DLOG.
+    // A wildcard entry in the map will signal DLOG is on, but all VLOG levels are disabled.
+    LOG(INFO) << "TVM_LOG_DEBUG enables DLOG statements only";
+    map.emplace(kDefaultKeyword, -1);
```

Comment:
this logic seems like not backward compatible in performance. when spec == "1", VerboseEnabledInMap still need to do string parse and map search, that seems like not necessary, is there any specific concern here not just leave map be empty and return false in VerboseEnabledInMap when map.empty() is true?

Reply:
Everything is gated on TVM_LOG_DEBUG, signalling this is a developer build for the purposes of debugging. If this were on prod code paths then we'd need to do some very special tricks to cache the lookup per file.

Reply:
TVM_LOG_DEBUG gate only can help for TVM_LOG_DEBUG not define or be 0 case, when #define TVM_LOG_DEBUG 1, and TVM_LOG_DEBUG="1",  VerboseLoggingEnabled should directly return false instead of still doing filename parse and map search, I think that a normal check instead of special trick and doing correct check should also apply for debug build.

Reply:
Done.